speechbrain/lobes/models/wav2vec.py

```diff
@@ -0,0 +1,610 @@
+        self.strides = strides
+        self.out_dim = out_channels[-1]
+        # ! Note this does conv, norm, gelu, dropout. while fairseq does conv, dropout, norm, gelu
+        # Also fairseq layernorm is forced to fp32
+        self.extractor = ConvolutionFrontEnd(
+            (None, 16000, 1,),
+            num_blocks=num_blocks,
+            num_layers_per_block=1,
+            out_channels=out_channels,
+            kernel_sizes=kernel_sizes,
+            strides=strides,
+            dilations=[1] * num_blocks,
+            residuals=[False] * num_blocks,
+            conv_module=Conv1d,
+            activation=nn.GELU,
+            norm=LayerNorm,
+            dropout=dropout,
+            conv_bias=False,
+            padding="valid",
+            conv_init=init,
+        )
+
+    def forward(self, x):
+        x = x.unsqueeze(2)
+        return self.extractor(x)
+
+    def get_output_lengths(self, input_lengths: torch.LongTensor):
+        def _conv_out_length(input_length, kernel_size, stride):
+            return torch.floor((input_length - kernel_size) / stride + 1)
+
+        for kernel_size, stride in zip(self.kernel_sizes, self.strides):
+            input_lengths = _conv_out_length(input_lengths, kernel_size, stride)
+        return input_lengths.to(torch.long)
+
+
+class W2VLatentProjector(nn.Module):
+    def __init__(self, extractor_dim, embedding_dim):
+        super().__init__()
+        self.norm = nn.LayerNorm(extractor_dim)
+        self.proj = nn.Linear(extractor_dim, embedding_dim)
+
+    def forward(self, x):
+        # non-projected features are needed as targets
+        normed_x = self.norm(x)
+        projed_normed_x = self.proj(normed_x)
+        return projed_normed_x, normed_x
+
+
+class Wav2VecPositionalConvEmbedding(nn.Module):
+    """Positional encoding as implemented in fairseq.
+    """
+
+    def __init__(self, embedding_dim, pos_conv_kernel=128, pos_conv_groups=16):
+        super().__init__()
+        self.embedding_dim = embedding_dim
+        self.pos_conv = nn.Conv1d(
+            self.embedding_dim,
+            self.embedding_dim,
+            kernel_size=pos_conv_kernel,
+            padding=pos_conv_kernel // 2,
+            groups=pos_conv_groups,
+        )
+        std = math.sqrt((4 * (1.0)) / (pos_conv_kernel * embedding_dim))
+        nn.init.normal_(self.pos_conv.weight, mean=0, std=std)
+        nn.init.constant_(self.pos_conv.bias, 0)
+
+        self.pos_conv = nn.utils.weight_norm(
+            self.pos_conv, name="weight", dim=2
+        )
+        self.pos_conv = nn.Sequential(
+            self.pos_conv, SamePadLayer(pos_conv_kernel), nn.GELU()
+        )
+
+    def forward(self, x):
+        x = x.transpose(1, 2)
+        x = self.pos_conv(x)
+        x = x.transpose(1, 2)
+        return x
+
+
+class W2VLatentEncoder(nn.Module):
+    """Wraps transformer and positional encoder.
+    """
+
+    def __init__(
+        self,
+        embedding_dim=768,
+        num_layers=8,
```

Comment:
```suggestion
        num_layers=12,
```
It is the number of layers in the wav2vec2.0 BASE model.